src/asl.ts

```diff
@@ -0,0 +1,1909 @@
+      ) {
+        return `$.0_${element.name}`;
+      } else {
+        throw new Error(
+          `cannot use an Identifier to index an Array or Object except for an array in a for-in statement`
+        );
+      }
+    }
+    return `${toJsonPath(expr.expr)}[${elementToJsonPath(expr.element)}]`;
+  }
+
+  function elementToJsonPath(expr: Expr): string {
+    if (expr.kind === "StringLiteralExpr") {
+      return `'${expr.value}'`;
+    } else if (expr.kind === "NumberLiteralExpr") {
+      return expr.value.toString(10);
+    }
+
+    throw new Error(
+      `Expression kind '${expr.kind}' is not allowed as an element in a Step Function`
+    );
+  }
+
+  export const isTruthy = (v: string): Condition =>
+    and(
+      isPresent(v),
+      isNotNull(v),
+      or(
+        and(isString(v), not(stringEquals(v, ""))),
+        and(isNumeric(v), not(numericEquals(v, 0))),
+        and(isBoolean(v), ref(v))
+      )
+    );
+
+  export const ref = (Variable: string): Condition => ({ Variable });
+
+  export const and = (...cond: Condition[]): Condition => ({
+    And: cond,
+  });
+
+  export const or = (...cond: Condition[]): Condition => ({
+    Or: cond,
+  });
+
+  export const not = (cond: Condition): Condition => ({
+    Not: cond,
+  });
+
+  export const isPresent = (Variable: string): Condition => ({
+    IsPresent: true,
+    Variable,
+  });
+
+  export const isNull = (Variable: string): Condition => ({
+    IsNull: true,
+    Variable,
+  });
+
+  export const isNotNull = (Variable: string): Condition => ({
+    IsNull: false,
+    Variable,
+  });
+
+  export const isBoolean = (Variable: string): Condition => ({
+    IsBoolean: true,
+    Variable,
+  });
+
+  export const isString = (Variable: string): Condition => ({
+    IsString: true,
+    Variable,
+  });
+
+  export const isNumeric = (Variable: string): Condition => ({
+    IsNumeric: true,
+    Variable,
+  });
+
+  export const stringEqualsPath = (
+    Variable: string,
+    path: string
+  ): Condition => ({
+    And: [
+      isString(Variable),
+      {
+        StringEqualsPath: path,
+      },
+    ],
+  });
+
+  export const stringEquals = (
+    Variable: string,
+    string: string
+  ): Condition => ({
+    And: [
+      isString(Variable),
+      {
+        StringEquals: string,
+      },
+    ],
+  });
+
+  export const numericEqualsPath = (
+    Variable: string,
+    path: string
+  ): Condition => ({
+    And: [
+      isNumeric(Variable),
+      {
+        NumericEqualsPath: path,
+      },
+    ],
+  });
+
+  export const numericEquals = (
+    Variable: string,
+    number: number
+  ): Condition => ({
+    And: [
+      isNumeric(Variable),
+      {
+        NumericEquals: number,
+      },
+    ],
+  });
+
+  export function toCondition(expr: Expr): Condition {
+    if (expr.kind === "BooleanLiteralExpr") {
+      return {
+        IsPresent: !expr.value,
+        Variable: `$.0_${expr.value}`,
+      };
+    } else if (expr.kind === "UnaryExpr") {
```

Comment:
Unary could also be a negative number. `-10`

Reply:
Right you are!

Reply:
https://github.com/sam-goodwin/functionless/issues/new